spec/rspec/mocks/partial_double_spec.rb

```diff
@@ -379,6 +390,22 @@ def defined_private_method
         object.send(:defined_private_method)
       end
 
+      it 'runs the before_verifying_double callbacks on any_instance' do
+        expect { |probe|
+          RSpec.configuration.mock_with(:rspec) do |config|
+            config.before_verifying_doubles(&probe)
+          end
+
+          expect_any_instance_of(klass).to receive(:implemented)
+          object.implemented
+        }.to yield_successive_args(
+          *(klass.ancestors + [klass]).map do |target|
```

Comment:
I don't think it's desirable.  Having the callback trigger 4+ times (for the class, `Object`, `Kernel`, and `BasicObject`) for a single use of `any_instance` is incredibly confusing.  We should redo the implementation so that it is only triggered once, with the specific class passed to `expect_any_instance_of`.  That fact that we instantiate a recorder for each ancestor is an implementation detail and should not be exposed here.